pkg/apis/nfd/v1alpha1/rule.go

```diff
@@ -97,7 +101,7 @@ func (r *Rule) Execute(features Features) (RuleOutput, error) {
 	ret := RuleOutput{Labels: labels, Vars: vars}
 	utils.KlogDump(2, fmt.Sprintf("rule %q matched with: ", r.Name), "  ", ret)
 
-	return ret, nil
+	return anyMatched, featureMatched, ret, nil
```

Comment:
Hmm, that will require `RuleOutput` struct to be updated for something that will not be used. Perhaps we can hook a warning somewhere else instead of sending the taints and then ignoring it?

Reply:
Iirc the idea behind ruleoutput was that it would contain all changes applied to the node (we'll likely have annotations in the future, too). The taints part would be used by nfd-master but not nfd-worker. Of course we could have something like a  `taints bool` arg on the Execute() function and then hook a warning somewhere else in nfd-worker code path, but imo that would feel just useless complication. Currently we don't print any warning on nfd-worker side (if taints are specified), I think.

Reply:
Do we really need to add this warning in the nfd-worker? I feel like, as long as the feature field doesn't exist in the worker config then user doesn't have a wrong assumption about availability of the feature. I mean if config doesn't allow me to set something then I assume it is not supported. WDYT?

Reply:
It might be useful if people mix up things. The thing is that the yaml/json unmarshalling (of the config) does not have any spec validation. Unknown fields are just silently ignored. People can put taints field and nfd-worker is happy with that,

Reply:
You have a point, agreed. Added a warning in https://github.com/kubernetes-sigs/node-feature-discovery/pull/910/commits/d451e5db712ddd133393fa842b206b1a70a5aa5e.